src/main/java/org/mskcc/kickoff/validator/AutoGenerabilityValidator.java

```diff
@@ -0,0 +1,53 @@
+package org.mskcc.kickoff.validator;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.mskcc.kickoff.domain.KickoffRequest;
+import org.mskcc.kickoff.notify.GenerationError;
+import org.mskcc.kickoff.printer.ErrorCode;
+import org.mskcc.kickoff.util.Constants;
+import org.mskcc.kickoff.util.Utils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+class AutoGenerabilityValidator {
+    private static final Logger PM_LOGGER = Logger.getLogger(Constants.PM_LOGGER);
+    private static final Logger DEV_LOGGER = Logger.getLogger(Constants.DEV_LOGGER);
+
+    private final ErrorRepository errorRepository;
+
+    @Autowired
+    public AutoGenerabilityValidator(ErrorRepository errorRepository) {
+        this.errorRepository = errorRepository;
+    }
+
+    public void validate(KickoffRequest kickoffRequest) {
```

Comment:
Not sure why it implements `validate(kickoffRequest)` instead of overriding `test(kickoffRequest)`. Will `validate(kickoffRequest)` be called in `RequestValidator.java`?

Reply:
is `validateShiny` not needed?